### PR TITLE
Simplify the code

### PR DESCRIPTION
**Pull request by Fyg369**
## Type
- [ ] bug fix 

- [ ] feature add 

- [ ] other 

## Checkbox
- [ ] do not have break changes 

- [ ] normal operation will not be affected after modification 

## Linked issues
<!-- if want to fix it, try "fixes: #13">

### CODE REVIEW

**Comment by dabao1955** [comment]
？

**Comment by Fyg369** [comment]
> ？

代码逻辑不变只是简化重复代码

**Comment by dabao1955** [comment]
> > ？
> 
> 代码逻辑不变只是简化重复代码，还有昨天合并后分支没有被删除

删的了

**Comment by Fyg369** [comment]
下面这段代码里的echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15语句,是否需要把clang不存在也进进语句里？因为i没有把if [ -d $HOME/clang ]; then不成立考虑进去

                 if [ ! -z ${{ inputs.other-gcc64-url }} ] && [ ! -z ${{ inputs.other-gcc32-url }} ]; then
                     if [ -d $HOME/clang ]; then
                         export OTHER_GCC64_URL=${{ inputs.other-gcc64-url }}
                         export OTHER_GCC32_URL=${{ inputs.other-gcc32-url }}
                         download_and_extract "$OTHER_GCC64_URL" "gcc-aarch64" "$HOME/gcc-64" "${{ inputs.other-gcc64-branch }}"
                         download_and_extract "$OTHER_GCC32_URL" "gcc-arm" "$HOME/gcc-32" "${{ inputs.other-gcc32-branch }}"
                     fi
                 else
                     echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15
                 fi

File: action.yml [hunk 1]
```diff
@@ -227,7 +227,6 @@ runs:
          if [ ${{ inputs.aosp-clang }} = false ] && [ ${{ inputs.android-ndk }} = false ]; then
              echo "::group:: Downloading Third party clang"
              export OTHER_CLANG_URL=${{ inputs.other-clang-url }}
-             mkdir -p -v $HOME/clang
              download_and_extract "$OTHER_CLANG_URL" "clang" "$HOME/clang" "${{ inputs.other-clang-branch }}"
              [ ! -d $HOME/clang/bin/ ] && mv $HOME/clang/*/* $HOME/clang/
          fi
```
Comment (dabao1955) [inline action.yml:232]: ？

**Review by dabao1955 — CHANGES_REQUESTED** [review]

**Comment by dabao1955** [comment]
> 下面这段代码里的echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15语句,是否需要把clang不存在也进进语句里？因为i没有把if [ -d $HOME/clang ]; then不成立考虑进去
> 
>                  if [ ! -z ${{ inputs.other-gcc64-url }} ] && [ ! -z ${{ inputs.other-gcc32-url }} ]; then
>                      if [ -d $HOME/clang ]; then
>                          export OTHER_GCC64_URL=${{ inputs.other-gcc64-url }}
>                          export OTHER_GCC32_URL=${{ inputs.other-gcc32-url }}
>                          download_and_extract "$OTHER_GCC64_URL" "gcc-aarch64" "$HOME/gcc-64" "${{ inputs.other-gcc64-branch }}"
>                          download_and_extract "$OTHER_GCC32_URL" "gcc-arm" "$HOME/gcc-32" "${{ inputs.other-gcc32-branch }}"
>                      fi
>                  else
>                      echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15
>                  fi

可以

File: action.yml (hunk 1 above)
Comment (Fyg369) [reply action.yml:232]: > ？

这个代码已经在函数里了

**Comment by Fyg369** [comment]
> > 下面这段代码里的echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15语句,是否需要把clang不存在也进进语句里？因为i没有把if [ -d $HOME/clang ]; then不成立考虑进去
> > ```
> >              if [ ! -z ${{ inputs.other-gcc64-url }} ] && [ ! -z ${{ inputs.other-gcc32-url }} ]; then
> >                  if [ -d $HOME/clang ]; then
> >                      export OTHER_GCC64_URL=${{ inputs.other-gcc64-url }}
> >                      export OTHER_GCC32_URL=${{ inputs.other-gcc32-url }}
> >                      download_and_extract "$OTHER_GCC64_URL" "gcc-aarch64" "$HOME/gcc-64" "${{ inputs.other-gcc64-branch }}"
> >                      download_and_extract "$OTHER_GCC32_URL" "gcc-arm" "$HOME/gcc-32" "${{ inputs.other-gcc32-branch }}"
> >                  fi
> >              else
> >                  echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15
> >              fi
> > ```
> 
> 可以

大佬语句怎么添加比较合适？请帮忙添加下   谢谢

File: action.yml (hunk 1 above)
Comment (dabao1955) [reply action.yml:232]: > > ？
> 
> 这个代码已经在函数里了

我说的是下一行clangurl那个

**Comment by dabao1955** [comment]
> > > 下面这段代码里的echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15语句,是否需要把clang不存在也进进语句里？因为i没有把if [ -d $HOME/clang ]; then不成立考虑进去
> > > ```
> > >              if [ ! -z ${{ inputs.other-gcc64-url }} ] && [ ! -z ${{ inputs.other-gcc32-url }} ]; then
> > >                  if [ -d $HOME/clang ]; then
> > >                      export OTHER_GCC64_URL=${{ inputs.other-gcc64-url }}
> > >                      export OTHER_GCC32_URL=${{ inputs.other-gcc32-url }}
> > >                      download_and_extract "$OTHER_GCC64_URL" "gcc-aarch64" "$HOME/gcc-64" "${{ inputs.other-gcc64-branch }}"
> > >                      download_and_extract "$OTHER_GCC32_URL" "gcc-arm" "$HOME/gcc-32" "${{ inputs.other-gcc32-branch }}"
> > >                  fi
> > >              else
> > >                  echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15
> > >              fi
> > > ```
> > 
> > 可以
> 
> 大佬语句怎么添加比较合适？请帮忙添加下   谢谢

添加后面url检测那里就行

**Comment by Fyg369** [comment]
> > > > 下面这段代码里的echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15语句,是否需要把clang不存在也进进语句里？因为没有把if [ -d $HOME/clang ]; then不成立考虑进去
> > > > ```
> > > >              if [ ! -z ${{ inputs.other-gcc64-url }} ] && [ ! -z ${{ inputs.other-gcc32-url }} ]; then
> > > >                  if [ -d $HOME/clang ]; then
> > > >                      export OTHER_GCC64_URL=${{ inputs.other-gcc64-url }}
> > > >                      export OTHER_GCC32_URL=${{ inputs.other-gcc32-url }}
> > > >                      download_and_extract "$OTHER_GCC64_URL" "gcc-aarch64" "$HOME/gcc-64" "${{ inputs.other-gcc64-branch }}"
> > > >                      download_and_extract "$OTHER_GCC32_URL" "gcc-arm" "$HOME/gcc-32" "${{ inputs.other-gcc32-branch }}"
> > > >                  fi
> > > >              else
> > > >                  echo "Error: Please enable third party other-gcc64-url and other-gcc32-url." && exit 15
> > > >              fi
> > > > ```
> > > 
> > > 
> > > 可以
> > 
> > 
> > 大佬语句怎么添加比较合适？请帮忙添加下   谢谢
> 
> 添加后面url检测那里就行

Please enable third party 'gcc64-url' and 'gcc32-url', and ensure that 'clang' directory exists.
这样可以吗？

File: action.yml (hunk 1 above)
Comment (Fyg369) [reply action.yml:232]: > > > ？
> > 
> > 
> > 这个代码已经在函数里了
> 
> 我说的是下一行clangurl那个

"$OTHER_CLANG_URL"

Comment (Fyg369) [reply action.yml:232]: 只是删除了创建其它没变动

**Comment by dabao1955** [comment]
if [ ! -z ${{ inputs.other-gcc64-url }} ] && [ ! -z ${{ inputs.other-gcc32-url }} ] && [ ! -d $HOME/clang ]; then

**Comment by Fyg369** [comment]
> if [ ! -z ${{ inputs.other-gcc64-url }} ] && [ ! -z ${{ inputs.other-gcc32-url }} ] && [ ! -d $HOME/clang ]; then

？

**Comment by Fyg369** [comment]
怎么删除ndk了？

**Comment by dabao1955** [comment]
> Please enable third party 'gcc64-url' and 'gcc32-url', and ensure that 'clang' directory exists.
> 这样可以吗？
可以

**Comment by Fyg369** [comment]
> > Please enable third party 'gcc64-url' and 'gcc32-url', and ensure that 'clang' directory exists.
> > 这样可以吗？
> 
> 可以

已修改，看下是否还有其它问题

**Review by dabao1955 — APPROVED** [review]

**Pull request merged**